src/components/Task.tsx

```diff
@@ -13,13 +13,20 @@ function hexToRgb(hex: string, alpha: number) {
 interface TaskProps {
     task: Database['public']['Tables']['Tasks']['Row'];
     handleClick: (task: Database['public']['Tables']['Tasks']['Row']) => void;
+    handleBoxClick: (task: Database['public']['Tables']['Tasks']['Row']) => void;
 }
 
-const TaskCard = ({ task, handleClick }: TaskProps) => {
+const TaskCard = ({ task, handleClick, handleBoxClick }: TaskProps) => {
     const [isComplete, setIsComplete] = useState(task.completed);
     const currentDate = new Date();
 
     const handleCheckboxClick = () => {
+        setIsComplete(!isComplete);
+        handleBoxClick(task);
+    }
+
+    // change this to edit task
+    const handleTaskClick = () => {
         setIsComplete(!isComplete);
```

Comment:
Why toggle isComplete when clicking the box of the task to edit?

Reply:
meant to change this. Done